apps/emqx_bridge_gcp_pubsub/src/emqx_bridge_gcp_pubsub_consumer_schema.erl

```diff
@@ -0,0 +1,251 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023-2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_bridge_gcp_pubsub_consumer_schema).
+
+-import(hoconsc, [mk/2, ref/2]).
+
+-include_lib("typerefl/include/types.hrl").
+-include_lib("hocon/include/hoconsc.hrl").
+
+%% `hocon_schema' API
+-export([
+    namespace/0,
+    roots/0,
+    fields/1,
+    desc/1
+]).
+
+%% `emqx_bridge_v2_schema' "unofficial" API
+-export([
+    source_examples/1,
+    conn_bridge_examples/1,
+    connector_examples/1
+]).
+
+-define(CONNECTOR_TYPE, gcp_pubsub_consumer).
+-define(SOURCE_TYPE, gcp_pubsub_consumer).
+
+%%-------------------------------------------------------------------------------------------------
+%% `hocon_schema' API
+%%-------------------------------------------------------------------------------------------------
+
+namespace() ->
+    "gcp_pubsub_consumer".
+
+roots() ->
+    [].
+
+%%=========================================
+%% Action fields
+%%=========================================
+fields(source) ->
+    {gcp_pubsub_consumer,
+        mk(
+            hoconsc:map(name, ref(?MODULE, consumer_source)),
+            #{
+                desc => <<"GCP PubSub Consumer Source Config">>,
+                required => false
+            }
+        )};
+fields(consumer_source) ->
+    emqx_bridge_v2_schema:make_consumer_action_schema(
+        mk(
+            ref(?MODULE, source_parameters),
+            #{
+                required => true,
+                desc => ?DESC(consumer_source)
+            }
+        ),
+        #{resource_opts_ref => ref(?MODULE, source_resource_opts)}
+    );
+fields(source_parameters) ->
+    %% FIXME: check
```

Comment:
Hide topic mapping?  How to add a single topic and combine that with the legacy mapping?